adoc/deployment-vmware.adoc

```diff
@@ -40,13 +40,43 @@ for the other method.
 [NOTE]
 ====
 If you choose the {ay} method, please ignore all the following steps for the
-VMWare template creation.
+VMware template creation.
 ====
 
 For each VM deployment, follow the {ay} installation method used for deployment on
 bare metal machines as described in <<deployment_bare_metal>>.
 
-=== Setup using VMWare Template
+=== Setup Using the VMware Template
+
+==== Choose a Disk Format for the Template
+
+Before creating the template, it is important to select the right format of the root hard disk for the node. This format is then used by default when creating new instances with Terraform.
+
+For the majority of cases, we recommend using *Thick Provision Lazy Zeroed*. This format is quick to create, provides good performance, and avoids the risk of running out of disk space due to over-provisioning.
+
+[NOTE]
+====
+It is not possible to resize a disk when using *Thick Provision Eager Zeroed*.
+For this reason, the Terraform variables `master_disk_size` and `worker_disk_size` must be set to the exact same size as in the original template.
+====
+
+link:https://pubs.vmware.com/vsphere-51/index.jsp?topic=%2Fcom.vmware.vsphere.storage.doc%2FGUID-4C0F4D73-82F2-4B81-8AA7-1DD752A8A5AC.html[Official VMware documentation] describes these formats as follows:
+
+Thick Provision Lazy Zeroed::
+Creates a virtual disk in a default thick format. Space required for the virtual disk is allocated when the disk is created. Data remaining on the physical device is not erased during creation, but is zeroed out on demand later on first write from the virtual machine. Virtual machines do not read stale data from the physical device.
+
+Thick Provision Eager Zeroed::
+A type of thick virtual disk that supports clustering features such as Fault Tolerance. Space required for the virtual disk is allocated at creation time. In contrast to the thick provision lazy zeroed format, the data remaining on the physical device is zeroed out when the virtual disk is created. It might take longer to create virtual disks in this format than to create other types of disks. Increasing the size of an Eager Zeroed Thick virtual disk causes a significant stun time for the virtual machine.
+
+Thin Provision::
+Use this format to save storage space. For the thin disk, you provision as much datastore space as the disk would require based on the value that you enter for the virtual disk size. However, the thin disk starts small and at first, uses only as much datastore space as the disk needs for its initial operations. If the thin disk needs more space later, it can grow to its maximum capacity and occupy the entire datastore space provisioned to it.
+
+[IMPORTANT]
+.Select the Disk Format Thoughtfully
+====
+It is not possible to change the format in Terraform after deployment. In order to switch to a different format, you would have to create a new template, which requires a new deployment.
```

Comment:
@lcavajani I changed this sentence, does it still make sense? or in other words what are the consequences of having to create a new TF template? Is it really a new deployment?

Reply:
@nkoranova `after deployment` is not correct, this is so before deploying.

The idea is: If you create a template with **Thin Provisioning**, then you can only create instances with terraform with **Thin Provisioning**, youu can't switch to **Thich Provisioning**.